src/containers/sound-editor.jsx

```diff
@@ -51,7 +59,11 @@ class SoundEditor extends React.Component {
             trimEnd: null
         });
     }
-    submitNewSamples (samples, sampleRate) {
+    submitNewSamples (samples, sampleRate, skipUndo) {
+        if (!skipUndo) {
+            this.redoStack = [];
+            this.undoStack.push(this.props.samples.slice(0));
```

Comment:
They can get large yes. I'd like to save that for a performance pass, when we can look into similar questions about max lengths of recordings.